### DISCUSSION TITLE
Date formats interpreted inconsistently

### DISCUSSION THREAD

**Discussion by johfc** [WinMerge/winmerge#1278] [Q&A]
It seems that WinMerge inconsistently applies date formats to text (eg in csv files) at least for the format aa-bb-cccc which implies the date format dd-mm-yyyy or mm-dd-yyyy if so interpreted. WinMerge will interpret either format with a preference for mm-dd-yyyy and only use dd-mm-yyyy when the former is invalid. This is inconsistent behaviour.

I have created the following files to demonstrate:

1. [Dates dd-mm-yyyy or mm-dd-yyyy.csv](https://github.com/WinMerge/winmerge/files/8396102/Dates.dd-mm-yyyy.or.mm-dd-yyyy.csv) - csv that contains a sample of text separated as follows: aa,bb,cccc,aa-bb-cccc
2. [Dates dd-mm-yyyy or mm-dd-yyyy.xlsx](https://github.com/WinMerge/winmerge/files/8396116/Dates.dd-mm-yyyy.or.mm-dd-yyyy.xlsx) - Excel spreadsheet (xlsx) from the csv file above
3. [Dates dd-mm-yyyy or mm-dd-yyyy - saved with Excel.csv](https://github.com/WinMerge/winmerge/files/8396117/Dates.dd-mm-yyyy.or.mm-dd-yyyy.-.saved.with.Excel.csv) - saved with Excel.csv" - csv saved in Excel to show how Excel modifies the csv contents!

If files 1 and 2 are compared in WinMerge, there are differences in the interpretation of implied date formats that are not apparent in the source files.

File 3 demonstrates how Excel modifies the contents of a csv file, which is unexpected.

**Comment by sdottaka** [comment]
Fixed this issue with commit 3782b0d9576456273c4f5addc11e5db4e6d733fd.

If you replace the C:\Program Files\WinMerge\MergePlugins\CompareMSExcelFiles.sct file with the following files, it will work as expected.

https://raw.githubusercontent.com/WinMerge/winmerge/master/Plugins/dlls/CompareMSExcelFiles.sct 

Thank you for the report.
